w1r3/infra/metrics/main.tf

```diff
@@ -0,0 +1,222 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+variable "metric-prefix" {
+    default = "ssb/w1r3"
+}
+
+resource "google_monitoring_metric_descriptor" "latency" {
+  description = "Operation latency as measured by the benchmark."
+  display_name = "Operation Latency"
+  type = "workload.googleapis.com/${var.metric-prefix}/latency"
+  metric_kind = "CUMULATIVE"
+  value_type = "DISTRIBUTION"
```

Comment:
I think a distribution gives us more detail.  We can get minute by minute distributions, like so:

![image](https://github.com/googleapis/storage-shared-benchmarking/assets/6241635/f7ea6061-3da4-48a0-8461-ab87a35acb80)

That gives us a sense of how "spread out" is the data in each time interval. And from histograms we can get percentiles too. We also get exemplars connecting the histograms to traces:

https://screenshot.googleplex.com/C9Q7nXbG349N2Tz.png

And I think it is easier to reason about how histograms aggregate when you have different attributes, or when you compute the p50 over longer time intervals.

Reply:
It occurs to me that you were suggesting we use "GAUGE" and "HISTOGRAM".  I am not sure there is a significant difference between the two [^1]. However, by default the exporters send cumulative data, so that seems easier to implement.

[^1] The computation of min/max ranges are easier with `GAUGE`.  But, AFAICT, those go unused by GCM exporters, so it does not matter.